### ISSUE TITLE
Add randomness

### ISSUE THREAD

**Issue by alxhoff** [munober/webscrapper#22]
**Is your feature request related to a problem? Please describe.**
The tool always downloads the same images. 

**Describe the solution you'd like**
Add some randomness, ie. skip a few pages on google images (not too many :P) or on imdb

**Comment by munober** [comment]
on it

**Comment by alxhoff** [comment]
also seems like the face detection fails sometimes :(

**Comment by munober** [comment]
yeah, i noticed that. what i did over the weekend is i compiled a dataset of 200k samples and filtered it for faces twice using opencv - first on the full size downloads, then on the cropped ones, and although it did reduce the size to around 140k, there still were things like neckties and weird shapes left that it took faces

**Comment by munober** [comment]
but we might need another tool for a double check or some smarter decision tree for what we keep and what we delete

**Comment by alxhoff** [comment]
hopefully it is good enough. Are there multiple models in openCV that you can apply?

**Comment by alxhoff** [comment]
also you can merge my two PRs now

**Comment by munober** [comment]
i am not entirely sure, but I will investigate. It uses dlib, which is based on that same 2015 Google FaceNet paper and there might be several .xml files you can download as network weights at least.

**Comment by munober** [comment]
> also you can merge my two PRs now

merged them

**Comment by alxhoff** [comment]
> i am not entirely sure, but I will investigate. It uses dlib, which is based on that same 2015 Google FaceNet paper and there might be several .xml files you can download as network weights at least.

If possible we should apply more than just one so we can clean up the data set as much as possible. Maybe make a generic function to filter on faces that takes in the weights data so we can just call it with a ton of models

**Comment by munober** [comment]
second face filter is done and it has not left a single non-face picture in the dataset, looks very promising, I'm using some guy's implementation of dlib off github.
I am not so sure anymore about the added randomness part though, because it might, on google, leave us with very few usable results. Look how a search for DiCaprio using the trademark free flag dries up into random crap after about 50 pictures: www.google.com/search?safe=off&site=&tbm=isch&source=hp&q=Leonardo_dicaprio&oq=Leonardo_dicaprio&gs_l=img&tbs=sur:fc

And it's worse for non-dicaprio's, on google. What do you think? Should we still do it on google?

But I will be adding randomness for imdb on the next commit.

**Comment by alxhoff** [comment]
hmmm I was wondering this. I'd say it's up to you to decide what we can achieve before loosing performance.

**Comment by munober** [comment]
ok so the way I handled it is I'm shuffling both page numbers and results on imdb and that still delivers nice results.
On google, I just randomize the order of the results but don't touch the pages. I realized that even if it very likely doesn't deliver pictures of that specific actor's face, it still mostly finds pictures of people, and since we mostly don't care about name labels (right?), I guess it should be fine like that.
